matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/terms/DefaultTermsService.kt

```diff
@@ -48,11 +51,49 @@ internal class DefaultTermsService @Inject constructor(
 
     override suspend fun getTerms(serviceType: TermsService.ServiceType,
                                   baseUrl: String): GetTermsResponse {
-        val url = buildUrl(baseUrl, serviceType)
-        val termsResponse = executeRequest(null) {
-            termsAPI.getTerms("${url}terms")
+        return if (serviceType == TermsService.ServiceType.Homeserver) {
+            getHomeserverTerms(baseUrl)
+        } else {
+            val url = buildUrl(baseUrl, serviceType)
+            val termsResponse = executeRequest(null) {
+                termsAPI.getTerms("${url}terms")
+            }
+            GetTermsResponse(termsResponse, getAlreadyAcceptedTermUrlsFromAccountData())
+        }
+    }
+
+    /**
+     * We use a trick here to get the homeserver T&C, we use the register API
+     */
+    private suspend fun getHomeserverTerms(baseUrl: String): GetTermsResponse {
+        return try {
+            executeRequest(null) {
+                termsAPI.register(baseUrl + NetworkConstants.URI_API_PREFIX_PATH_R0 + "register")
```

Comment:
is this something we should consider adding to the spec? it looks like we're relying on the `register` endpoint to fail

Reply:
the register endpoint will fail for sure, we are not providing any data to the request. Also if registration is disabled, I'm not sure what will happen.
This is a temporary workaround waiting for https://github.com/matrix-org/matrix-doc/pull/3012 to land